### PR TITLE
Updated markup in plugins to be more ADA compliant

### PR DESCRIPTION
**Pull request by sharissis-brei**
In order to complete some ADA compliance issues I found, I had to alter the markup generated by some of the slider/carousel plugins.

### CODE REVIEW

**Comment by tatwell** [comment]
@sharissis-brei I just spoke to @mandythomas. We will pull your branch, deploy your changes to our staging server, and let you know when they are ready for testing. This will enable you to test them. We will test them as well. Mandy is expecting to receive a doc today from Erika outlining the changes. Once she receives that, we'll schedule the change.

Looking at the code changes here in the pull request, my major questions are:

- Will `carousel.js` still work after the removal of the `href` lines?
- How was the `flexsider.js` plugin updated? My understanding is that this is a 3rd party library. Did we pull an updated version of the plugin or make the changes ourselves? If we did make the changes ourselves, how are the changes documented so that we will be aware of them in the future?

Once the changes here are deployed to the staging server, I'll post an update here.

**Comment by sharissis-brei** [comment]
Thank you, that sounds great. To address your questions:

- I can't be certain if carousel.js will still work after I remove the hrefs, as I don't have a place to run it locally to test. If there's a way to do this, please let me know and I can try it. Otherwise, I was hoping we could test it in the QA CMS and revert if it breaks.
- The diff here is hard to read because it's minified, but I added the following to the generated slider navigation links:
`href="#" title="Move to the previous slide"`
`href="#" title="Move to the next slide"`

**Comment by tatwell** [comment]
The latest version of [flexslider.js](https://github.com/woothemes/FlexSlider) appears to be v2.6.3. We are using (v2.2.0)[https://github.com/sharissis-brei/cascade-assets/blob/da2bd0695521c97f5f93a251cc276474e00a8920/app/assets/javascripts/cascade/plugins/flexslider.js]. 

I think the more conservative approach to making this change would be to first update to the latest version of FlexSlider (unminified version). If that doesn't fix it, then we could make the change ourselves -- in the unminified version (this will be easier to review). We don't need to include the minified version here in our repo as our deploy script will do the minifying. 

If we wanted to be good open source citizens, we could also issue a PR to the FlexSlider repository as there is no reason why they shouldn't prefer ADA compliance. I imagine they also have a test suite which these changes could be tested against.

**Comment by sharissis-brei** [comment]
I'll remove the href changes in flexslider.js for now, and then we can update the plugin and go from there.

Would you like me to update the old file with the newest version, or is there another way we should go about this? I know you have a certain include/build system and want to make sure I'm following the right process.

**Comment by tatwell** [comment]
@sharissis-brei I'm going to talk to @mandythomas shortly, so let me double-check with her. My recommendation would be to delete the min file and add the latest version of the unminified library to the project. But I'll post an update here this afternoon after I talk to Mandy.

In the meantime, you might wish to check to see if the change is still necessary with the new version. You should be able to test that outside Cascade.

**Comment by sharissis-brei** [comment]
I took a look through the unminified current version of flexslider: https://github.com/woothemes/FlexSlider/blob/master/jquery.flexslider.js.

The good news is that the new version has href values! The bad news is there are still not title attributes on any of the navigation. Therefore, we will want to add those manually.

**Comment by tatwell** [comment]
Drats! 

Out of curiosity, is it possible to add the title values on document load? Will that meet compliance expectation? The advantage is we could leave the library code as is and even make the fix compatible with future versions of FlexSider that do add the title.

If not, we can proceed with changing the new unminified version. I would recommend a comment at top of file indicating the change we make.

Thanks!

**Comment by sharissis-brei** [comment]
That's a good idea, but we might have trouble getting the timing right if we take that approach – the script may execute before the slider is finished loading unless we set a timeout or something.

From a best practices standpoint, I'd personally vote to add the titles to the markup in the plugin. It's your cal though. Let me know how you'd like me to proceed.

**Comment by tatwell** [comment]
@sharissis-brei I'm ok with updating the plugin. 👍

**Comment by tatwell** [comment]
@sharissis-brei Mandy and I will meet tomorrow at 11am to release this branch to staging server. I'll notify you once release is complete.

**Comment by sharissis-brei** [comment]
Thank you @tatwell !

**Comment by tatwell** [comment]
@mandythomas and I deployed updated assets from this branch to dev-cascade and tested carousel changes here:

http://dev-www.chapman.edu/campus-services/marketing-communication/guidelines-and-resources/web/modular-widgets/2-column-modular-example-page.aspx

Carousel is working! 👍 

We're still looking at other changes.

**Comment by mandythomas** [comment]
Sponsor widget is working. As seen on:
http://dev-www.chapman.edu/campus-services/marketing-communication/guidelines-and-resources/web/modular-widgets/primary-logo-image-rotator.aspx 

http://dev-www.chapman.edu/campus-services/marketing-communication/guidelines-and-resources/web/modular-widgets/right-logo-image-rotator.aspx 

faculty.js, faculty-law, directory.js relate to various faculty directory listing pages but those are not migrated YET to use the master.js and therefore the changes in github won't effect those. The live pages use same js from local versions inside Cascade. However same ADA fixes have been made to those versions, to automate AltText for photos using person's name from database

**Comment by mandythomas** [comment]
The flexslider is working. It's used by the Slider masthead as seen on http://dev-www.chapman.edu/campus-services/marketing-communication/guidelines-and-resources/web/modular-widgets/masthead-example-slider.aspx. I don't know if it's used for other widgets.

**Comment by tatwell** [comment]
@mandythomas Thanks for update. Let me know when you are ok to merge branch into master.

@sharissis-brei Will you be doing your own UA testing on staging server to verify ADA compliance? Can you let me know when your testing is complete and you're ok to merge PR?

I notice the updated version of FlexSlider never got added. I'm ok leaving it as is now that we've tested the changes and confirmed it works. But please add a comment at top of flexslider.js that notes what changes were made to original code along with a TODO to update to latest version.

**Comment by sharissis-brei** [comment]
@tatwell @mandythomas Yes, we will be doing testing on the staging server to confirm the ADA compliance issues we've logged. If there are any outstanding issues, I will add commits to this PR. Otherwise, I'll let you know when we've completed testing so you can merge the PR.

As for Flexslider, perhaps I misunderstood our previous conversation. I thought you had mentioned adding it, but if you'd like me to, I'd be happy to update it. However, there's always the change a new version would not be stable with the slider instantiations/markup that's in place now, so keeping it as is isn't a bad idea in my opinion. Let me know what you'd like me to do.

File: app/assets/javascripts/cascade/plugins/flexslider.js [hunk 1]
```diff
@@ -3,4 +3,9 @@
  * Copyright 2012 WooThemes
  * Contributing Author: Tyler Smith
  */
+
+/* CHANGES:
+* Added descriptive title attributes to paging navigation - sharissis-brei
```
Comment (tatwell) [inline app/assets/javascripts/cascade/plugins/flexslider.js]: 👍  Yes, this is what I had in mind. Can you date it and explain the why behind change, like this (I'd do it myself but I don't have access to your repo):

`2016-09-13: Added descriptive title attributes to paging navigation for ADA compliance. - sharissis-brei`

Also, please include this TODO below it:

```
TODO(tatwell): Before any future changes, update to latest version. Right now we're at 
2.2.0. Current version (https://github.com/woothemes/FlexSlider) is 2.6.3. May need to 
redo ADA-compliance changes we made to this version. 
```

We're trying to get a little more rigorous with our JS dev practices over here and code comments and maintainability are a focus. Thanks!

**Comment by tatwell** [comment]
@sharissis-brei Thanks for updates. Yes, I'm ok with keeping current version of flexslider.

**Comment by mandythomas** [comment]
Sarah I am fairly sure that app/assets/stylesheets/widgets/masthead.scss is not the one being used on the live site. It 's a duplicate. The one still in use is app/assets/stylesheets/layout_old/phase4_masthead.scss. Not sure if you want to make same change there or we should or ?? @tatwell

**Comment by mandythomas** [comment]
Sarah I have finished reviewing. Besides notes above, not sure how to know if some others are working or adversely affecting anythiing else... mostly those stylesheet changes for no-js.  I did publish out the whole site to dev-www.chapman.edu yesterday. The pages linked from this page are examples of many widgets and mastheads so I don't know if you want to take a look at some of those to see if you feel they're working as you think? 
http://dev-www.chapman.edu/campus-services/marketing-communication/guidelines-and-resources/web/modular-widgets/index.aspx 
@tatwell

**Comment by tatwell** [comment]
@sharissis-brei @mandythomas When you are each done with your testing, please comment here with a thumbs-up 👍  and I'll proceed with merge. Thanks!

**Comment by sharissis-brei** [comment]
@mandythomas I can make the change to the masthead file. @tatwell I'll let you know when we've finished our testing.

**Comment by mandythomas** [comment]
just fyi, the version of cascade-assets (latest master.css and js) on dev-www.chapman.edu is no longer the one with the ADA updates, it's Mirabel's latest from a different branch so on some newly republished pages we're seeing "Jump to the main content of this page" text that is meant to be hidden. When Mirabel is done with her changes, if Tom or Sarah want to see the site with the ADA pull request version then it will need to be reuploaded into dev-cascade by Tom and pages published out again. I will be at a conference most of next week

**Comment by sharissis-brei** [comment]
@mandythomas @tatwell We were going to do some testing early this week. Is there any way we could redeploy my changes? Let me know, thanks!

**Comment by tatwell** [comment]
@sharissis-brei I'll redeploy your branch as soon as I have a chance today. I'll update you when it's ready.

**Comment by tatwell** [comment]
@sharissis-brei I've re-released this branch to the staging server. You should be able to resume your testing. Let me know if you encounter any issues.

**Comment by sharissis-brei** [comment]
Thank you @tatwell !

**Comment by sharissis-brei** [comment]
Hi @tatwell and @mandythomas, are the changes from this PR currently on the staging server? We are doing some QA resolution today and I want to make sure everything's in place, thanks.

**Comment by tatwell** [comment]
Yes, @sharissis-brei, staging server should currently be serving this code. 

We have another release on hold until this is done so if you can let us know once you've completed your testing, I'd appreciate it.

**Comment by sharissis-brei** [comment]
Thanks for confirming @tatwell. I had to add a couple things just now – would you mind deploying from the latest once more?

**Comment by sharissis-brei** [comment]
Hi @mandythomas and @tatwell , question for you regarding the Flexslider plugin. Is there another version of flexslider.js somewhere that may be part of your build? I ask because I've made changes to flexslider.js in this repo (adding titles to links), but the changes don't seem to be taking effect.

Let me know if you have any insight on this, thanks.

**Comment by mandythomas** [comment]
what page are you viewing it on (just in case the issue is that the latest assets or this page are not published out? If that doesn't turn out to be it I'll do a search of the github files to see where else it comes up.

**Comment by sharissis-brei** [comment]
http://dev-www.chapman.edu/admission/undergraduate/index.aspx

**Comment by mandythomas** [comment]
@sharissis-brei You can ignore that admission undergraduate page. On the DEV server that page is using an old "Department" template. We've retired it but the content on the DEV server is fairly old at this point so doesn't reflect the retirement of that template. On production that page is now using one of our Modulare templates. Same is true of all the school/dept homepages eg /dodge or /law etc. They are using the old "School and Dept" template and that's now retired and those pages in production use the new 1-Column Modular.   @tatwell , is there a 1 Column Modular page with all the widgets on it that can be used as a good test page on DEV? Something similiar to http://dev-www.chapman.edu/campus-services/marketing-communication/guidelines-and-resources/web/modular-widgets/2-column-modular-example-page.aspx ??

**Comment by sharissis-brei** [comment]
Thanks for the info @mandythomas . Let us know if there's a better page to test Flexslider on. In the meantime, let me know if you find any other instances of the Flexslider JS that may need to be altered.

**Comment by mandythomas** [comment]
This page has a Slider masthead which uses flexslider.js
http://dev-www.chapman.edu/campus-services/marketing-communication/guidelines-and-resources/web/modular-widgets/masthead-example-slider.aspx

**Comment by tatwell** [comment]
@sharissis-brei Any update on this PR? Are you still testing? Do you need anything from me?

**Comment by sharissis-brei** [comment]
Hi @tatwell, we've completed most of our testing and are ready to merge this in when you are.

**Comment by tatwell** [comment]
Thanks @sharissis-brei.

@homeofmatt @mirabelrice, please review this pull request. Once you approve, I'll merge it.

**Comment by mirabelrice** [comment]
👍

**Comment by homeofmatt** [comment]
Looks good!

**Pull request merged**